### PR TITLE
Adds a coder override emote so manual custom emotes work properly

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

Doing emote("me" some text) expected the emote to be player fired. It's being used in the code as if it's a "just do it you fuck" tool. So I added a "Just do it you fuck" tool.

I've also updated some truly stupid remnants of what I assume to be old emote systems from parrot code.

## Why It's Good For The Game

Makes "me" emotes on mobs actually work, lets #52594 go through sanely

## Changelog
:cl:
fix: Lots of mob emotes will actually work now. Observe your local Runtime today 
/:cl:

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
Oh wait I might need to check how this is fired don't I

**Comment by LemonInTheDark** [comment]
it sucks but it works

**Comment by Jared-Fogle** [comment]
I'm not a fan of the magic and repetitiveness of `.emote("code", ..., hardcoded = TRUE)`. Might I recommend adding a slight layer of abstraction to it by adding something like `/mob/proc/fluff_emote` or something similar, which would just run what you have now but not require the user to explicitly put in "code" and the hardcoded flag?

**Comment by LemonInTheDark** [comment]
that does sound nice, I was somewhat unsure how to solve this issue, I'll give it a shot.

File: code/datums/emotes.dm [hunk 1]
```diff
@@ -157,3 +157,40 @@
 		var/mob/living/L = user
 		if(HAS_TRAIT(L, TRAIT_EMOTEMUTE))
 			return FALSE
+
+/datum/emote/coders_screech //This is a shorthand to allow for custom emotes on our side while ignoring any player only checks that come with "me"
+	key = "code"
+
+/datum/emote/coders_screech/run_emote(mob/user, params, type_override, intentional = FALSE, hardcoded = FALSE) //Just override the song and dance
```
Comment (Rohesie) [inline code/datums/emotes.dm:164]: Why a `hardcoded` parameter if you are already making a new emote type?
You can just redefine the checks there.

File: code/datums/emotes.dm [hunk 2]
```diff
@@ -157,40 +157,36 @@
...
-
-/datum/emote/coders_screech/run_emote(mob/user, params, type_override, intentional = FALSE, hardcoded = FALSE) //Just override the song and dance
+/**
+* Allows the intrepid coder to send a basic emote
+* Takes text as input, sends it out to those who need to know after some light parsing
+* If you need something more complex, make it into a datum emote
+* Arguments:
+* * text - The text to send out
+*/
+/mob/proc/manuel_emote(text) //Just override the song and dance
```
Comment (Rohesie) [inline code/datums/emotes.dm:167]: is `manuel` instead of `manual` here intended?

Comment (LemonInTheDark) [reply code/datums/emotes.dm]: Whatever do you mean?

**Pull request merged**